core/Util/plJPEG.cpp

```diff
@@ -189,6 +189,15 @@ void plJPEG::DecompressJPEG(hsStream* S, void* buf, size_t size) {
         offs += out_stride;
     }
 
+    // Data stored as RGB on disk but Plasma uses BGR
+    uint32_t* dp = reinterpret_cast<uint32_t*>(buf);
```

Comment:
As in the PlasmaShop PR, we should probably add an assert to ensure the `buf` is properly aligned for the cast...

Reply:
libHSPlasma doesn't seem to have a standard assertion macro like Plasma does, so I added a `throw hsBadParamException` if the alignment doesn't smell right.

Reply:
Yeah, that's reasonable...